src/pages/mealPlanPage/MealPlanPage.jsx

```diff
@@ -0,0 +1,109 @@
+import { useLocation } from 'react-router-dom';
+import React, { useEffect, useState } from 'react';
+import ApiQuery from '../../components/shared/api/ApiQuery';
+import { Carousel, Container, Row } from 'react-bootstrap';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faClock } from '@fortawesome/free-solid-svg-icons';
+import TimeFormatted from '../../components/shared/timeFormatted/TimeFormatted';
+import { EditControls } from '../../components/shared/editControls/EditControls';
+import { ROUTES_PATHS } from '../../routes';
+import './mealPlanPage.css';
+import RecipeTile from '../../components/shared/tiles/recipeTile/RecipeTile';
+
+function MealPlanPage(props) {
+    const { isLoggedIn, isOwner } = props;
+
+    const location = useLocation();
+
+    const [mealPlanData, setMealPlanData] = useState('');
+    const [recipes, setRecipes] = useState();
+
+    const fetchData = async () => {
+        try {
+            setMealPlanData((await ApiQuery.get(`mealPlans/${location.state.id}`)).data);
```

Comment:
I would reccomend to move this await thing about and assign it to a variable and then pass variable as an argument to useState function. The same fot setRecipies